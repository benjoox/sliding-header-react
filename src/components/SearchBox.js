import React from 'react'
import { bool } from 'prop-types'
import { ReactComponent as Search } from './assets/search.svg'

export default function SearchBox({ collapsed }) {
    console.log('collapsed ', collapsed)
  return (
    <div className="search-box-container">
      <div className={`search-box ${collapsed ? 'collapsed' : ''}`}>
        <div className={`search-icon ${collapsed ? 'collapsed' : ''}`}>
          <Search />
        </div>
        <input type="text" className="search-box-input" placeholder="Search" />
      </div>
    </div>
  )
}

SearchBox.protoType = {
  collap: bool.isRequired,
}
