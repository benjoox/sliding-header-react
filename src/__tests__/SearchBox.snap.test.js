import React from 'react'
import renderer from 'react-test-renderer'
import SearchBox from '../components/SearchBox'

it('SearchBox renders correctly when collpased', () => {
  const tree = renderer.create(<SearchBox collapsed={true}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('SearchBox renders correctly when expanded', () => {
    const tree = renderer.create(<SearchBox collapsed={false}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
