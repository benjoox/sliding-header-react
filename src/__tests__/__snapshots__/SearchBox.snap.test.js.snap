// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly when collpased 1`] = `
<div
  className="search-box-container"
>
  <div
    className="search-box collapsed"
  >
    <div
      className="search-icon collapsed"
    >
      <svg>
        search.svg
      </svg>
    </div>
    <input
      className="search-box-input"
      placeholder="Search"
      type="text"
    />
  </div>
</div>
`;

exports[`SearchBox renders correctly when expanded 1`] = `
<div
  className="search-box-container"
>
  <div
    className="search-box "
  >
    <div
      className="search-icon "
    >
      <svg>
        search.svg
      </svg>
    </div>
    <input
      className="search-box-input"
      placeholder="Search"
      type="text"
    />
  </div>
</div>
`;
